CONTRIBUTING.md

```diff
@@ -1,3 +1,189 @@
+  the changes, and provides feedback, edits, and suggestions. Check out the
+  [open pull requests](https://github.com/jupyterhub/kubespawner/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc)
+  and provide feedback that helps improve the PR and get it merged. Please keep your
+  feedback positive and constructive!
+* **Tell people about kubespawner.**
+  Kubespawner is built by and for its community.
+  If you know anyone who would like to use kubespawner, please tell them about the project!
+  You could give a talk about it or run a demonstration or make a poster.
+  The sky is the limit :rocket::star2:.
+
+
+## Setting up for documentation changes
+
+We use Sphinx to build the kubespawner documentation. You can make changes to
+the documentation with any text editor and directly through the GitHub website.
+
+For small changes (like typos) you do not need to setup anything locally. For
+larger changes we recommend you build the documentation locally so you can see
+the end product in its full glory.
+
+To make edits through the GitHub website visit https://github.com/jupyterhub/kubespawner/tree/master/docs/source, open the file you would like to edit and then click "edit". GitHub will
+walk you through the process of proposing your change ("making a Pull Request").
+
+A brief guide to setting up for local development
+```sh
+git clone https://github.com/jupyterhub/kubespawner.git
+cd kubespawner/docs
+# create a new environment ...
+conda create -n kubespawner --file environment.yml
+# ... or update your environment
+conda env update --file environment.yml
+# activate the environment
+conda activate kubespawner
+# build the documentation
+make html
+```
+
+## Setting up a local development environment
+
+To work on kubespawner's code you can run JupyterHub locally on your computer,
+using an editable installation of kubespawner, that interacts with pods in a
+local kubernetes cluster!
+
+You need to have a local kubernetes cluster and be able to edit networking
+rules on your computer. We will now walk you through the steps to get going:
+
+1.  Install VirtualBox by [downloading and running an
+    installer](https://www.virtualbox.org/wiki/Downloads).
+
+1.  Install
+    [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/).
+
+1.  Run `minikube start`. This will start your kubernetes cluster if it isn't
+    already up. Run `kubectl get node` to make sure it is.
+
+    Note that the `minikube start` command will also setup `kubectl` on your
+    host machine to interact with the kubernetes cluster along with a
+    `~/.kube/config` file with credentials for connecting to this cluster.
+
+1.  Setup a networking route so that a program on your host can talk to the
+    pods inside minikube.
+
+    ```bash
+    # Linux
+    sudo ip route add 172.17.0.0/16 via $(minikube ip)
+
+    # MACOS
+    sudo route -n add -net 172.17.0.0/16 $(minikube ip)
+    # later on you can undo this it with
+    sudo route delete -net 172.17.0.0
```

Comment:
added this command

Reply:
if someone knows what the linux equivalent is we can add that above as well

Reply:
The man page says:
```
ip route { add | del | change | append | replace } ROUTE
```
So, I believe the "undo" linux equivalent would be
```
sudo ip route del 172.17.0.0/16
```

Reply:
Thanks for finding out, I added it.